docs/src/appendix/scarb-toml.md

```diff
@@ -0,0 +1,248 @@
+
+Adjust Cairo compiler configuration parameters when compiling this package. These options are not taken into consideration when this package is used as a dependency for another package. All fields are optional.
+
+```toml
+[profile.dev.cairo]
+# ...
+```
+
+#### `unstable-add-statements-code-locations-debug-info`
+See [`unstable-add-statements-code-locations-debug-info`](https://docs.swmansion.com/scarb/docs/reference/manifest.html#unstable-add-statements-code-locations-debug-info) in Scarb documentation.
+
+```toml
+[profile.dev.cairo]
+unstable-add-statements-code-locations-debug-info = true
+```
+
+#### `unstable-add-statements-functions-debug-info`
+See [`unstable-add-statements-functions-debug-info`](https://docs.swmansion.com/scarb/docs/reference/manifest.html#unstable-add-statements-functions-debug-info) in Scarb documentation.
+```toml
+[profile.dev.cairo]
+unstable-add-statements-functions-debug-info = true
+```
+
+#### `inlining-strategy`
+See [`inlining-strategy`](https://docs.swmansion.com/scarb/docs/reference/manifest.html#inlining-strategy) in Scarb documentation.
+```toml
+[profile.dev.cairo]
+inlining-strategy = "avoid"
+```
+
+#### Example of configuration which allows [coverage](https://foundry-rs.github.io/starknet-foundry/testing/coverage.html) report generation
+```toml
+[profile.dev.cairo]
+unstable-add-statements-code-locations-debug-info = true
+unstable-add-statements-functions-debug-info = true
+inlining-strategy = "avoid"
+```
+
+### `[features]`
+A package defines a set of named features in the `[features]` section of `Scarb.toml` file. Each defined feature can list other features that should be enabled with it. All fields are optional.
+```toml
+[features]
+# ...
+```
+
+#### `<feature-name>`
+The `<feature-name>` field specifies the name of the feature and list of other features that should be enabled with it.
+See [features](https://docs.swmansion.com/scarb/docs/reference/conditional-compilation.html#features) in Scarb documentation.
+```toml
+[features]
+enable_for_tests = []
+```
+
+#### Example of `Scarb.toml` allowing conditional contracts compilation
+Firstly, define a contract in the src directory with a `#[cfg(feature: '<FEATURE_NAME>')]` attribute:
+```rust
+#[starknet::contract]
+#[cfg(feature: 'enable_for_tests')]
+mod MockContract {
+    // ...
+}
+```
+
+Then update Scarb.toml so it includes the following lines:
+```toml
+[features]
+enable_for_tests = []
+```
+
+### `[[target.starknet-contract]]`
+Scarb supports registering targets that are handled by Scarb extensions. Such targets are called external. All fields are optional. See more about [targets](https://docs.swmansion.com/scarb/docs/reference/targets.html) in Scarb documentation.
+
```

Comment:
Let's just focus on `starknet-contract` target and say that this is required to build starknet contracts and link to this https://docs.swmansion.com/scarb/docs/extensions/starknet/contract-target.html#starknet-contract-target page. 

`sierra` and `casm` can just link to relevant subsections.

Reply:
Ok, looks as below now:
<img width="823" alt="image" src="https://github.com/user-attachments/assets/79d25282-cd18-49b7-a55e-05f3ded4ab85">